rest-rest/src/Components/RestStep.tsx

```diff
@@ -0,0 +1,696 @@
+  FileUpload,
+  InputGroup,
+  Checkbox,
+  FormSelect,
+  FormSelectOption,
+  Popover,
+  Grid,
+  GridItem,
+} from '@patternfly/react-core';
+import { Endpoint } from './Endpoint';
+import { OpenAPI } from 'openapi-types';
+import { useEffect, useState } from 'react';
+import SwaggerParser from '@apidevtools/swagger-parser';
+import { parse } from 'yaml';
+import { IStepProps } from '../../../try-catch-eip/kaoto/types/dts/src/types.js';
+import MimeTypes from './MimeTypes';
+import { HelpIcon, TrashIcon, PlusCircleIcon } from '@patternfly/react-icons';
+
+export interface IEndpoint {
+  name: string;
+  path: string;
+  operations:
+  | Map<string, OpenAPI.Operation>
+  | Map<string, { operationId: string; }>;
+  produces: Map<string, string[]>;
+  consumes: Map<string, string[]>;
+  produce: Map<string, string>;
+  consume: Map<string, string>;
+}
+
+async function parseApiSpec(
+  input: string | OpenAPI.Document
+): Promise<IEndpoint[]> {
+  let swaggerParser: SwaggerParser = new SwaggerParser();
+
+  const e: Array<IEndpoint> = [];
+  try {
+    await swaggerParser.validate(input, { dereference: { circular: 'ignore' } });
+    const paths = swaggerParser.api.paths;
+    if (paths) {
+      Object.keys(paths).forEach((key: string) => {
+        const path = paths[key];
+        if (path) {
+          const operations: Map<string, OpenAPI.Operation> = new Map<string, OpenAPI.Operation>();
+          Object.entries(path).forEach((method: [string, OpenAPI.Operation]) => {
+            operations.set(method[0], method[1]);
+          });
+          let produces: Map<string, string[]> = new Map<string, string[]>();
+          let consumes: Map<string, string[]> = new Map<string, string[]>();
+          let produce: Map<string, string> = new Map<string, string>();
+          let consume: Map<string, string> = new Map<string, string>();
+          operations.forEach((op: OpenAPI.Operation, verb: string) => {
+            const producesMediaType: string[] = [];
+            if ('produces' in op) {
+              op.produces?.forEach((prod: string) => producesMediaType.push(prod));
+              if (producesMediaType.length > 0) {
+                produce.set(verb, producesMediaType[0]);
+              }
+            } else if (op.responses) {
+              Object.values(op.responses).forEach((response) => {
+                if (response.content) {
+                  producesMediaType.push.apply(producesMediaType, (Object.keys(response.content)));
+                }
+              });
+            }
+            produces.set(verb, producesMediaType);
+            if ('consumes' in op) {
+              const consumesMediaTypes: string[] = [];
+              consumes.set(verb, consumesMediaTypes);
+              op.consumes?.forEach((con: string) => consumesMediaTypes.push(con));
+              if (consumesMediaTypes.length > 0) {
+                consume.set(verb, consumesMediaTypes[0]);
+              }
+            }
+            //@ts-ignore
```

Comment:
:cry: can we workaround this somehow to avoid using `@ts-ignore`?

Reply:
Don't know how, it complains because `verb` is a string so it is not sure what is going to be the type there.

We know what the type is going to be. Typescript doesn't.